src/crimson/os/seastore/cache.cc

```diff
@@ -1173,7 +1173,13 @@ void Cache::complete_commit(
     bool is_inline = false;
     if (i->is_inline()) {
       is_inline = true;
-      i->set_paddr(final_block_start.add_relative(i->get_paddr()));
+      if (final_block_start.get_addr_type() == addr_types_t::SEGMENT) {
+	i->set_paddr(final_block_start.add_relative(i->get_paddr()));
+      } else if (final_block_start.get_addr_type() ==
+	addr_types_t::RANDOM_BLOCK) {
+	i->set_paddr(final_block_start.add_offset(
+	  i->get_paddr().as_seg_paddr().get_segment_off()));
+      }
```

Comment:
I think we need to discuss this. I suppose that there are two ways to use blk_paddr with existing seastore implementation.

1) allocate blk_paddr directly after epm.alloc_new_extent
2) update blk_paddr after submit_record 

IMO, 1) probably causes racing problem (performance degradation) and does not guarantee sequential allocation within a transaction context because alloc_new_extent can be called by multiple threads. So, without real allocation at alloc_new_extent(), it seems that address allocation after transaction is made is better way. Am I wrong?

Reply:
That allocation pathway is for ool extents.  It seems to me that extents within the CBJournal should be addressed pre-commit as with a SegmentedJournal -- as record-relative.  Post commit, CBJournal will return the actual blk_paddr addresses and we'll update the extent addresses as we do for any journal commit.

Reply:
Ok. The commit (seastore: enable cbjournal when running unittest_transaction_manager) works as your comment.